tests/kernel/wave/wave_e2e_test.py

```diff
@@ -60,3 +60,117 @@ def test(
     ):
         test(a, b)
         assert_allclose(a, b)
+
+
+@require_e2e
+def test_transpose_read():
+    M = tkl.sym.M
+    N = tkl.sym.N
+    BLOCK_M = tkl.sym.BLOCK_M
+    BLOCK_N = tkl.sym.BLOCK_N
+    ELEMS_PER_THREAD = tkl.sym.ELEMS_PER_THREAD
+    ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
+
+    constraints: list[tkw.Constraint] = [
+        tkw.HardwareConstraint(
+            threads_per_wave=64,
+            waves_per_block=(1, 1, 1),
+            vector_shapes={M: 1, N: BLOCK_N},
+        )
+    ]
+    constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
+    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 0)]
+    constraints += [tkw.WaveConstraint(M, BLOCK_M)]
+    constraints += [tkw.WaveConstraint(N, BLOCK_N)]
+
+    i = tkw.IndexMapping.iterator(0)
+    j = tkw.IndexMapping.iterator(1)
+    mapping = tkw.IndexMapping(
+        num_iterators=2, inputs={M: i, N: j}, outputs={M: i, N: j}

```

Comment:
NIT: Can we make `inputs={N: i, M: j}, outputs={N: i, M: j}` and set `a: tkl.Memory[M, N, ..` and `b: tkl.Memory[N, M...` to make the map more obviously transpose

Reply:
done